### ISSUE TITLE
String Validator for digits greater than PHP_INT_MAX doesn't work

### ISSUE THREAD

**Issue by galoppi** [thephpleague/json-guard#119]
The rule
```
"telephone": 
{
   "type": "string"
}
```
doesn't validate for 
```
{
   "telephone":"12372163870"
}
```
even though "telephone" is of type string in JSON string.
@see: vendor/league/json-guard/src/Constraints/Type.php:56

Why do you do this check here? Shouldn't this check be done for type int instead of string?

`json_decode($this->request->getRawBody(), false, 512, JSON_BIGINT_AS_STRING)` doesn't change the behaviour.

**Comment by yuloh** [comment]
Which version do you have installed?  This should be fixed in [1.0.1](https://github.com/thephpleague/json-guard/blob/master/CHANGELOG.md#101---2017-05-03).

**Comment by galoppi** [comment]
Oh, I see, yeah, with 1.0.1 it works.
Sorry!
